packages/jellyfish-api-core/__tests__/category/icxorderbook/submitDFCHTLC.test.ts

```diff
@@ -0,0 +1,328 @@
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import {
+  ExtHTLC, HTLC, ICXClaimDFCHTLCInfo, ICXDFCHTLCInfo, ICXEXTHTLCInfo, ICXGenericResult,
+  ICXListHTLCOptions, ICXOfferInfo, ICXOrderInfo, ICXOffer, ICXOrder, InputUTXO
+} from '../../../src/category/icxorderbook'
+import BigNumber from 'bignumber.js'
+import {
+  setup, accountDFI, idDFI, accountBTC, checkBTCBuyOfferDetails, checkBTCSellOrderDetails,
+  checkDFIBuyOfferDetails, checkDFISellOrderDetails, checkDFCHTLCDetails, checkEXTHTLCDetails
+} from './common.test'
+import { RpcApiError } from '../../../src'
+
+describe('Test ICXOrderBook.submitDFCHTLC', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+    await setup(container)
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  afterEach(async () => {
+    // cleanup code here
+  })
+
+  // common code for some tests until ICX offer
+  const setupUntilDFIBuyOffer = async (): Promise<string> => {
+    // create order - maker
+    const order: ICXOrder = {
+      tokenFrom: idDFI,
+      chainTo: 'BTC',
+      ownerAddress: accountDFI,
+      receivePubkey: '037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941',
+      amountFrom: new BigNumber(15),
+      orderPrice: new BigNumber(0.01)
+    }
+
+    let result: ICXGenericResult = await client.icxorderbook.createOrder(order, [])
+    const createOrderTxId = result.txid
+
+    await container.generate(1)
+
+    // list ICX orders
+    let orders: Record<string, ICXOrderInfo| ICXOfferInfo> = await client.icxorderbook.listOrders()
+    await checkDFISellOrderDetails(container, order, createOrderTxId, orders as Record<string, ICXOrderInfo>)
+
+    // make Offer to partial amout 10 DFI - taker
+    const offer: ICXOffer = {
+      orderTx: createOrderTxId,
+      amount: new BigNumber(0.10), // 0.10 BTC = 10 DFI
+      ownerAddress: accountBTC
+    }
+    const accountBTCBeforeOffer = await container.call('getaccount', [accountBTC, {}, true])
+
+    result = await client.icxorderbook.makeOffer(offer, [])
+    const makeOfferTxId = result.txid
+    await container.generate(1)
+
+    const accountBTCAfterOffer = await container.call('getaccount', [accountBTC, {}, true])
+
+    // check fee of 0.01 DFI has been reduced from the accountBTCBeforeOffer[idDFI]
+    // Fee = takerFeePerBTC(inBTC) * amount(inBTC) * DEX DFI per BTC rate
+    // NOTE(surangap): why sometimes garbage values are in expected, floting point representation problems?
```

Comment:
sometimes after an arithmetic calculation, the result(expected) has decimal garbage values. like following situations
```
Expected: 109.95700000000001
Received: 109.967
```

Floating-point representation problems?